core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -144,14 +159,17 @@ private[spark] class SparkUpgradeException(
 private[spark] class SparkArithmeticException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
It is tested by at least QueryExecutionErrorsSuite "CAST_OVERFLOW: from long to ANSI intervals"